File: api-create-commit-author/src/logger.py
import datetime as __date


__PURPLE = '\033[95m'
__CYAN = '\033[96m'
__BLUE = '\033[94m'
__GREEN = '\033[92m'
__YELLOW = '\033[93m'
__RED = '\033[91m'

__BOLD = '\033[1m'
__UNDERLINE = '\033[4m'
__END = '\033[0m'


def __get_time():
    return str(__date.datetime.now().strftime('%d/%m/%Y %H:%M:%S'))


def info(text: str):
    dt = __get_time()
    print(f'{dt} - {__BOLD}{__GREEN}[info]{__END}: {text}')


def error(text: str):
    dt = __get_time()
    print(f'{dt} - {__BOLD}{__RED}[error]{__END}: {text}')


def warn(text: str):
    dt = __get_time()
    print(f'{dt} - {__BOLD}{__YELLOW}[warn]{__END}: {text}')


def debug(text: str):
    dt = __get_time()
    print(f'{dt} - {__BOLD}[debug]{__END}: {text}')


def topic(topic, text: str):
    dt = __get_time()
    print(f'{dt} - {__BOLD}{__BLUE}[{topic}]{__END}: {text}')

File: api-notification/src/app/lib/logger/Logger.ts
/* eslint-disable no-console */
import date from '../../utils/date';

class Logger {
  private reset = '\x1b[0m';
  private bright = '\x1b[1m';

  private black = '\x1b[30m';
  private red = '\x1b[31m';
  private green = '\x1b[32m';
  private yellow = '\x1b[33m';
  private blue = '\x1b[34m';
  private magenta = '\x1b[35m';
  private cyan = '\x1b[36m';
  private white = '\x1b[37m';

  info(message?: string) {
    console.log(
      `${this._get_date()} - ${this.bright}${this.green}[info]${this.reset}:`,
      message,
    );
  }

  warn(message?: string) {
    console.log(
      `${this._get_date()} - ${this.bright}${this.yellow}[warn]${this.reset}:`,
      message,
    );
  }

  error(message?: string) {
    console.log(
      `${this._get_date()} - ${this.bright}${this.red}[error]${this.reset}:`,
      message,
    );
  }

  debug(message?: string) {
    console.log(
      `${this._get_date()} - ${this.bright}${this.white}[white]${this.reset}:`,
      message,
    );
  }

  private _get_date() {
    return date.format(new Date());
  }
}

export default new Logger();
